mixed-persistence/mixed-persistence-java-sbt-postgres/hello-impl/src/main/java/com/lightbend/lagom/recipes/mixedpersistence/hello/impl/readside/Greetings.java

```diff
@@ -0,0 +1,111 @@
+        // database with changes made to persistent entities.
+        readSide.register(UserGreetingRecordWriter.class);
+    }
+
+    /**
+     * Asynchronously queries the read-side database for a list of all greetings that have been set for any HelloEntity.
+     *
+     * @return a {@link CompletionStage} that completes with a list of all stored greetings
+     */
+//    public CompletionStage<PSequence<UserGreeting>> all() {
+//        return jpaSession
+//                .withTransaction(this::selectAllUserGreetings)
+//                .thenApply(TreePVector::from);
+//    }
+
+//    private List<UserGreeting> selectAllUserGreetings(EntityManager entityManager) {
+//        return entityManager
+//                .createQuery(SELECT_ALL_QUERY, UserGreeting.class)
+//                .getResultList();
+//    }
+
+    /**
+     * Event processor that handles {@link GreetingMessageChanged} events
+     * by writing {@link UserGreetingRecord} rows to the read-side database table.
+     */
+    static class UserGreetingRecordWriter extends ReadSideProcessor<HelloEvent> {
+        private final JpaReadSide jpaReadSide;
+
+        @Inject
+        UserGreetingRecordWriter(JpaReadSide jpaReadSide) {
+            this.jpaReadSide = jpaReadSide;
+        }
+
+        @Override
+        public ReadSideHandler<HelloEvent> buildHandler() {
+            return jpaReadSide.<HelloEvent>builder("UserGreetingRecordWriter")
+                    .setGlobalPrepare(entityManager -> createSchema())
+                    .setEventHandler(GreetingMessageChanged.class, this::processGreetingMessageChanged)
+                    .build();
+        }
+
+        private void createSchema() {
+            // This is a convenience for creating the read-side table in development mode.
+            // It relies on a Hibernate-specific property to provide idempotent schema updates.
+            Persistence.generateSchema("default",
+                    ImmutableMap.of("hibernate.hbm2ddl.auto", "update")
+            );
+        }
+
+        private void processGreetingMessageChanged(EntityManager entityManager, GreetingMessageChanged greetingMessageChanged) {
+          System.out.println("^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^processGreetingMessageChanged");
+          System.out.println(greetingMessageChanged);
+//            UserGreetingRecord record = entityManager.find(UserGreetingRecord.class, greetingMessageChanged.getName());
+//            if (record == null) {
+//                record = new UserGreetingRecord();
+//                record.setId(greetingMessageChanged.getName());
+//                entityManager.persist(record);
+//            }
+//            record.setMessage(greetingMessageChanged.getMessage());
+
+          UserGreetingRecord record = new UserGreetingRecord();
+          record.setId(greetingMessageChanged.getName());
+          record.setMessage(greetingMessageChanged.getMessage());
+          entityManager.persist(record);
```

Comment:
Is the comment a left-over? 

Also, does PGsql merge records when inserting data for the same PK? The commented code does a `find` first to update if the record is already there, but the new code doesn't.

If that's the case I think it's a very important difference compared to H2 worth mentioning in the README.